src/app/model/country-report.model.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from "@angular/core"
+import { Adapter } from "../model/adapter"
+
+export interface CountryApi {
+    Country: string //
+    CountryCode: string
+    Slug: string
+    NewConfirmed: number
+    TotalConfirmed: number
+    NewDeaths: number
+    TotalDeaths: number
+    NewRecovered: number
+    TotalRecovered: number
+    Date: string
+}
+
+export class CountryReport {
```

Comment:
Why are you using a class here, instead of an interface?

Reply:
I choose to use a class because I need to instantiate countryReport every time a user ask for country data in the this.countries array from the covid.service. 
When I do that, I'm mapping the properties from the original country object to my countryReport object, thus reducing the impact that possible changes made at server level could have in my template's code